core/controllers/topics_and_skills_dashboard.py

```diff
@@ -384,41 +423,25 @@ def post(self):
                 topic_services.add_uncategorized_skill(
                     self.user_id, topic.id, new_skill_id)
 
-        skill_domain.Skill.require_valid_description(description)
-
         if skill_services.does_skill_with_description_exist(description):
             raise self.InvalidInputException(
                 'Skill description should not be a duplicate.')
 
         skill = skill_domain.Skill.create_default_skill(
             new_skill_id, description, rubrics)
 
-        skill.update_explanation(
-            state_domain.SubtitledHtml.from_dict(explanation_dict))
+        skill.update_explanation(explanation_dict)
 
         image_filenames = skill_services.get_image_filenames_from_skill(skill)
 
         skill_services.save_new_skill(self.user_id, skill)
 
-        image_validation_error_message_suffix = (
-            'Please go to oppia.org/skill_editor/%s to edit '
-            'the image.' % skill.id)
         for filename in image_filenames:
-            image = self.request.get(filename)
-            if not image:
-                logging.exception(
-                    'Image not provided for file with name %s when the skill '
-                    'with id %s was created.' % (filename, skill.id))
-                raise self.InvalidInputException(
-                    'No image data provided for file with name %s. %s'
-                    % (filename, image_validation_error_message_suffix))
-            try:
-                file_format = (
-                    image_validation_services.validate_image_and_filename(
-                        image, filename))
-            except utils.ValidationError as e:
-                e = '%s %s' % (e, image_validation_error_message_suffix)
-                raise self.InvalidInputException(e)
+            image = files.get(filename)
+            image = base64.decodebytes(image.encode('utf-8'))
```

Comment:
In general, don't redeclare variables. Call this and the above variable separate things.

Reply:
Fixed it now.